src/test/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolderTest.java

```diff
@@ -265,6 +272,99 @@ public void runningBuild() throws Exception {
         d.assertItemNames(5);
     }
 
+    @Issue("JENKINS-60677")
+    @Test
+    public void runningBuildWithAbortBuildsOption() throws Exception {
+        SampleComputedFolder folder = r.jenkins.createProject(SampleComputedFolder.class, "d");
+        DefaultOrphanedItemStrategy strategy = new DefaultOrphanedItemStrategy(true, -1, -1);
+        strategy.setAbortBuilds(true);
+        folder.setOrphanedItemStrategy(strategy);
+        folder.kids.addAll(Arrays.asList("A", "B"));
+        folder.recompute(Result.SUCCESS);
+        folder.assertItemNames(1, "A", "B");
+        folder.getItem("B").getBuildersList().add(new SleepBuilder(Long.MAX_VALUE));
+        folder.getItem("B").setConcurrentBuild(true);
+        FreeStyleBuild bBuild1 = folder.getItem("B").scheduleBuild2(0).waitForStart();
+        FreeStyleBuild bBuild2 = folder.getItem("B").scheduleBuild2(0).waitForStart();
+        folder.kids.remove("B");
+        folder.recompute(Result.SUCCESS);
+        folder.assertItemNames(2, "A");
+        for (FreeStyleBuild bBuild : new FreeStyleBuild[] {bBuild1, bBuild2}) {
+            assertFalse(bBuild.isBuilding());
+            r.assertBuildStatus(Result.ABORTED, r.waitForCompletion(bBuild));
+            InterruptedBuildAction interruptedBuildAction = bBuild.getAction(InterruptedBuildAction.class);
+            CauseOfInterruption causeOfInterruption = interruptedBuildAction.getCauses().stream().findFirst().orElseThrow(NoSuchElementException::new);
+            assertTrue(causeOfInterruption instanceof OrphanedParent);
+        }
+        folder.recompute(Result.SUCCESS);
+        folder.assertItemNames(3, "A");
+        FreeStyleBuild aBuild1 = folder.getItem("A").scheduleBuild2(0).get();
+        FreeStyleBuild aBuild2 = folder.getItem("A").scheduleBuild2(0).get();
+        aBuild1.keepLog(true);
+        folder.kids.remove("A");
+        folder.recompute(Result.SUCCESS);
+        folder.assertItemNames(4, "A");
+        aBuild1.keepLog(false);
+        folder.recompute(Result.SUCCESS);
+        folder.assertItemNames(5);
+    }
+
+    @Issue("JENKINS-60677")
+    @Test
+    public void runningWorflowJobBuildWithAbortBuildsOption() throws Exception {
+        SampleComputedFolderWithWorkflowJobAsChildren folder = r.jenkins.createProject(SampleComputedFolderWithWorkflowJobAsChildren.class, "d");
+        DefaultOrphanedItemStrategy strategy = new DefaultOrphanedItemStrategy(true, -1, -1);
+        strategy.setAbortBuilds(true);
+        folder.setOrphanedItemStrategy(strategy);
+        folder.kids.add("kid");
+        folder.recompute(Result.SUCCESS);
+        folder.assertItemNames(1, "kid");
+        folder.getItem("kid").setDefinition(new CpsFlowDefinition("node { sleep(Long.MAX_VALUE) }", true));
```

Comment:
Or just

```suggestion
        folder.getItem("kid").setDefinition(new CpsFlowDefinition("sleep(Long.MAX_VALUE)", true));
```

should suffice I think.

And does this really work at all? You do not have a test dep on `workflow-basic-steps` nor `workflow-durable-task-step`.

If you add a test dep on the `tests` classified of `workflow-support` then you get the `semaphore` test step which is handy for things like this: can have the test wait for the build to get a well-defined point and pause before proceeding.

Reply:
You are correct, the build was doomed to fail but couldn't because abort was too fast.
I fixed that with the `semaphore` step.